sdk/tables/azure-data-tables/tests/test_table_entity.py

```diff
@@ -1221,7 +1221,7 @@ def test_operations_on_entity_with_partition_key_having_single_quote(self, table
 
             # Act
             sent_entity = self._create_updated_entity_dict(entity['PartitionKey'], entity['RowKey'])
-            resp = self.table.upsert_entity(mode=UpdateMode.MERGE, entity=sent_entity)
+            resp = self.table.upsert_entity(mode=UpdateMode.REPLACE, entity=sent_entity)
```

Comment:
Why change the UpdateMode here and some places below?

Reply:
Excellent question!
The original tests were always broken - and had been failing for some time, and we were skipping them because we _thought_ the reason they were failing was because this character wasn't supported by the service. In fact it was because we had a bug, and once that bug was fixed, the tests needed to be fixed to also run.